wgpu-hal/src/dx12/mod.rs

```diff
@@ -527,9 +528,21 @@ struct SwapChain {
     size: wgt::Extent3d,
 }
 
+struct DCompState {
+    visual: DirectComposition::IDCompositionVisual,
+    device: DirectComposition::IDCompositionDevice,
+    // Must be kept alive but is otherwise unused after initialization.
+    _target: DirectComposition::IDCompositionTarget,
+}
+
 enum SurfaceTarget {
     /// Borrowed, lifetime externally managed
     WndHandle(Foundation::HWND),
+    /// `handle` is borrowed, lifetime externally managed
+    VisualFromWndHandle {
+        handle: Foundation::HWND,
+        dcomp_state: RwLock<Option<DCompState>>,
```

Comment:
I think if we architect this (in a separate module) with a struct that looks like

```rust
#[derive(Default)]
struct DCompState {
    inner: RwLock<Option<InnerState>>,
}

impl DCompState {
    // Methods for init and deinit and such
}
```

This can help keep the separation of concerns pretty clear and have this code not need to think too hard about the dcomp integration.